packages/flutter/lib/src/cupertino/magnifier.dart

```diff
@@ -0,0 +1,308 @@
+      bounds: Rect.fromLTRB(
+          screenRect.left + widget.horizontalScreenEdgePadding,
+          // iOS doesn't reposition for Y, so we should expand the threshold
+          // so we can send the whole magnifier out of bounds if need be.
+          screenRect.top -
+              (CupertinoMagnifier.kDefaultSize.height +
+                  CupertinoMagnifier.kMagnifierAboveFocalPoint),
+          screenRect.right - widget.horizontalScreenEdgePadding,
+          screenRect.bottom +
+              (CupertinoMagnifier.kDefaultSize.height +
+                  CupertinoMagnifier.kMagnifierAboveFocalPoint)),
+      rect: rawMagnifierPosition & CupertinoMagnifier.kDefaultSize,
+    ).topLeft;
+
+    setState(() {
+      _currentAdjustedMagnifierPosition = adjustedMagnifierPosition;
+      // The lens should always point to the center of the line.
+      _verticalFocalPointAdjustment =
+          verticalPositionOfLens - verticalCenterOfCurrentLine;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedPositioned(
+      duration: CupertinoTextMagnifier._kDragAnimationDuration,
+      curve: widget.animationCurve,
+      left: _currentAdjustedMagnifierPosition.dx,
+      top: _currentAdjustedMagnifierPosition.dy,
+      child: CupertinoMagnifier(
+        inOutAnimation: _ioAnimation,
+        additionalFocalPointOffset: Offset(0, _verticalFocalPointAdjustment),
+      ),
+    );
+  }
+}
+
+/// A [RawMagnifier] in the Cupertino style.
+///
+/// [CupertinoMagnifier] is a wrapper around [RawMagnifier] that handles styling
+/// and transitions.
+///
+/// {@macro flutter.widgets.magnifier.intro}
+///
+/// See also:
+///
+/// * [RawMagnifier], the backing implementation.
+/// * [CupertinoTextMagnifier], a widget that positions [CupertinoMagnifier] based on
+/// [MagnifierOverlayInfoBearer].
+/// * [MagnifierController], the controller for this magnifier.
+class CupertinoMagnifier extends StatelessWidget {
+  /// Creates a [RawMagnifier] in the Cupertino style.
+  ///
+  /// The default constructor parameters and constants were eyeballed on
+  /// an iPhone XR iOS v15.5.
+  const CupertinoMagnifier({
+    super.key,
+    this.size = kDefaultSize,
+    this.borderRadius = const BorderRadius.all(Radius.elliptical(60, 50)),
+    this.additionalFocalPointOffset = Offset.zero,
+    this.shadows = const <BoxShadow>[
+              BoxShadow(
+                color: Color.fromARGB(25, 0, 0, 0),
+                blurRadius: 11,
+                spreadRadius: 0.2,
+              )
+            ],
+    this.borderSide =
+        const BorderSide(color: Color.fromARGB(255, 232, 232, 232)),
+    this.inOutAnimation,
+  });
+
+  /// The shadows displayed under the magnifier.
+  final List<BoxShadow> shadows;
+
+  /// The border, or "rim", of this magnifier.
+  final BorderSide borderSide;
+
+  /// The vertical offset, that the magnifier is along the Y axis above
+  /// the focal point.
+  @visibleForTesting
```

Comment:
unfortunately a visibleForTesting field is public in the eyes of the linter. Without it it would complain, so I put some intro level doc comments if anyone stumbles across it